ql/src/java/org/apache/hadoop/hive/ql/io/HiveInputFormat.java

```diff
@@ -260,6 +275,18 @@ public byte[] getBytesForHash() {
         return allBytes;
       }
     }
+
+    public byte[] getBytesForIdentity() {
```

Comment:
maybe 'getBytesForEquality'

Reply:
Renamed.
https://github.com/apache/hive/pull/5409/commits/6de1cf7be0d724877fc2cf98a00fb95f138d8924

BTW, do you know why we shift the offset [here](https://github.com/apache/hive/blob/6de1cf7be0d724877fc2cf98a00fb95f138d8924/ql/src/java/org/apache/hadoop/hive/ql/io/HiveInputFormat.java#L274)? If we can remove the `>> 3` or we are sure we can distinguish splits even with `>> 3`, we can simply use `getBytesForHash`. If we are not sure, I think we can keep this method for now for safety

```
SerDeUtils.writeLong(allBytes, pathBytes.length, getStart() >> 3);
```

Reply:
originally I wanted to ask why don't we reuse `getBytesForHash` and then noticed that we divide getStart() by 8
{code}
   // Explicitly using only the start offset of a split, and not the length. Splits generated on
    // block boundaries and stripe boundaries can vary slightly. Try hashing both to the same node.
    // There is the drawback of potentially hashing the same data on multiple nodes though, when a
    // large split is sent to 1 node, and a second invocation uses smaller chunks of the previous
    // large split and send them to different nodes.
{code}